apps/remix-ide/src/app/tabs/web3-provider.js

```diff
@@ -21,27 +21,38 @@ export class Web3ProviderModule extends Plugin {
     Should be taken carefully and probably not be release as it is now.
   */
   sendAsync (payload) {
-    return new Promise((resolve, reject) => {
-      const provider = this.blockchain.web3().currentProvider
-      // see https://github.com/ethereum/web3.js/pull/1018/files#diff-d25786686c1053b786cc2626dc6e048675050593c0ebaafbf0814e1996f22022R129
-      provider[provider.sendAsync ? 'sendAsync' : 'send'](payload, async (error, message) => {
-        if (error) return reject(error)
-        if (payload.method === 'eth_sendTransaction') {
-          if (payload.params.length && !payload.params[0].to && message.result) {
-            setTimeout(async () => {
-              const receipt = await this.tryTillReceiptAvailable(message.result)
-              if (!receipt.contractAddress) {
-                console.log('receipt available but contract address not present', receipt)
-                return
+    return this.askUserPermission('sendAsync', `Calling ${payload.method} with parameters ${JSON.stringify(payload.params)}`).then(
```

Comment:
it might be better readable to include `askUserPermission` in the promise callback.

Reply:
are you sure this is returning a promise?